packages/libluajit/build.sh

```diff
@@ -2,79 +2,52 @@ TERMUX_PKG_HOMEPAGE=https://luajit.org/
-		"${api_url}"
+	local latest_version current_version="${TERMUX_PKG_VERSION#*:}"
+	latest_version="$(
+		TERMUX_PKG_SRCURL="https://gitlab.archlinux.org/archlinux/packaging/packages/luajit" \
+		termux_gitlab_api_get_tag
 	)"
-	latest_version="$(jq -r '.results[0].pkgver' <<< "$response")"
-	unix_timestamp_latest="${latest_version##*.}"
-
-	if ! date -d "@${unix_timestamp_latest}" &> /dev/null; then
-		local summary
-		# shellcheck disable=SC2016
-		printf -v summary '%s\n' \
-			'### `libluajit`' \
-			''\
-			"Failed to get latest version of 'luajit'" \
-			"from '${api_url}'" \
-			''\
-			"Timestamp - $(date -d "@${unix_timestamp_latest}" --utc '+%Y-%m-%dT%H:%M:%SZ' 2>&1)" \
-			'Prettified `curl` response:' \
-			'```json' \
-			"$(jq -r <<< "$response")" \
-			'```'
-
-		if [[ -n "${GITHUB_STEP_SUMMARY:-}" ]]; then
-			# If this is the CI, output to the summary file
-			echo "$summary" >> "$GITHUB_STEP_SUMMARY"
-		else
-			# Otherwise output to whatever qualifies as FD2.
-			echo "$summary" >&2
-		fi
-		echo "WARN: Couldn't query new version. Staying at version '${TERMUX_PKG_VERSION}'."
-		return 0
-	fi
 
-	# If this isn't a dry-run add a human readable (ISO 8601)
-	# version of the timestamp as a comment to the version.
-	if [[ "${BUILD_PACKAGES}" != "false" && "$current_version" != "$latest_version" ]]; then
-		sed \
-			-e "s|^\(TERMUX_PKG_VERSION=.*\"\).*|\1 # $(date -d "@${unix_timestamp_latest}" --utc '+%Y-%m-%dT%H:%M:%SZ')|" \
-			-i "$TERMUX_PKG_BUILDER_DIR/build.sh"
-	fi
+	# Remove the -${rev} from the tag
+	latest_version="${latest_version%-*}"
 
-	termux_pkg_upgrade_version "${latest_version}"
+	# Use +g${hash} for the commit portion of the version
+	termux_pkg_upgrade_version "${latest_version/+/+g}"
 }
 
 termux_step_post_get_source() {
-	# Do the same as e.g. Arch linux is doing:
-	# The patch version is the timestamp of the above git commit, obtain via `git log`
-	local commit_timestamp="${TERMUX_PKG_VERSION:6}"
+	local expected_commit="${TERMUX_PKG_VERSION##*+g}"
+	local expected_timestamp="${TERMUX_PKG_VERSION##*.}"
+	expected_timestamp="${expected_timestamp%+g*}"
 
 	# Remember to pull in the necessary amount of git history
-	git fetch --shallow-since="$commit_timestamp"
+	git fetch --shallow-since="$expected_timestamp"
+
+	local actual_commit actual_timestamp
+	actual_timestamp="$(git show --no-patch --pretty=format:%at "$expected_commit" --)"
+	actual_commit="$(git log --date=unix --before="$expected_timestamp" --after="$expected_timestamp" --pretty=format:"%H")"
+
+	if [[ "${actual_commit::7}" != "${expected_commit::7}" ]]; then
+		termux_error_exit <<-EOF
+
+			ERROR: Expected and observed commit at timestamp don't match.
+			Expected: '${expected_commit::7}' @$expected_timestamp ($(date -d "@$expected_timestamp" --utc '+%Y-%m-%dT%H:%M:%SZ'))
+			Got:      '${actual_commit::7}' @$actual_timestamp ($(date -d "@$actual_timestamp" --utc '+%Y-%m-%dT%H:%M:%SZ'))
+		EOF
+	fi
```

Comment:
This uses the same check as before to get the timestamp of the commit, but it ensures that the actual timestamp and commit match their expected values from the version string.